Content.Client/Fax/FaxVisualsSystem.cs

```diff
@@ -0,0 +1,31 @@
+using Robust.Client.GameObjects;
+using Content.Shared.Fax;
+
+namespace Content.Client.Paint
+{
+    public sealed class FaxVisualsSystem : EntitySystem
+    {
+        /// <summary>
+        /// Visualizer for Paint which applies a shader and colors the entity.
+        /// </summary>
+        [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<FaxMachineComponent, AppearanceChangeEvent>(OnAppearanceChanged);
+        }
+
+        private void OnAppearanceChanged(EntityUid uid, FaxMachineComponent component, ref AppearanceChangeEvent args)
+        {
+            if (args.Sprite == null)
+                return;
+
+            if (_appearance.TryGetData<bool>(uid, FaxMachineVisuals.VisualState, out bool inserted))
+            {
+                args.Sprite.LayerSetState(FaxMachineVisuals.VisualState, component.InsertingState);
+            }
+        }
```

Comment:
This bool doesn't correspond to the enum data being sent from the server so not sure what this is even supposed to be (is it supposed to just play an animation instead?

Reply:
well if i remove the bool part the animation just loops forever. With it the animation plays once. Not sure why.

Reply:
That doesn't do anything.

1. You're setting it to both a bool and an enum on the server so the data is just garbage and doesn't correspond to anything.
2. This will just permanently set the state to this. If you want it to play an animation use animationplayer.

From the sounds of it you want a different enum entry and then check for that enum entry here, then use animationplayer system to play the animation.

Reply:
I changed it so no more out bool.